heareval/tasks/pipeline.py

```diff
@@ -556,6 +557,119 @@ def run(self):
         self.mark_complete()
 
 
+class ResampleSubcorpuses(WorkTask):
+    """
+    Aggregates resampling of all the splits and sampling rates
+    into a single task as dependencies.
+
+    Parameter:
+        metadata (ExtractMetadata): task which extracts a corpus level metadata
+    Requires:
+        subsample_splits (list(SubsampleSplit)): task subsamples each split
+    """
+
+    sample_rates = luigi.Parameter()
+    metadata = luigi.TaskParameter()
+
+    def requires(self):
+        # Perform resampling on each split and sampling rate independently
+        resample_splits = [
+            ResampleSubCorpus(
+                sr=sr,
+                split=split["name"],
+                metadata=self.metadata,
+                data_config=self.data_config,
+            )
+            for sr in self.sample_rates
+            for split in self.data_config["splits"]
+        ]
+        return resample_splits
+
+    def run(self):
+        workdir = Path(self.workdir)
+        workdir.rmdir()
+        # We need to link the workdir of the requires, they will all be the same
+        # for all the requires so just grab the first one.
+        requires_workdir = Path(self.requires()[0].workdir).absolute()
+        workdir.symlink_to(requires_workdir)
+        self.mark_complete()
+
+
+class GenerateAudioStats(WorkTask):
+    """
+    Generates the statistics of the extracted and sampled audio files
+    Parameters
+        metadata (ExtractMetadata): task which extracts a corpus level metadata
+    Requires
+        traintestmeta(SplitTrainTestMetadata): task which produces the split
+            level metadata
+    """
+
+    sample_rates = luigi.Parameter()
+    metadata = luigi.TaskParameter()
+
+    def requires(self):
+        return {
+            "resample": ResampleSubcorpuses(
+                sample_rates=self.sample_rates,
+                metadata=self.metadata,
+                data_config=self.data_config,
+            ),
+            "metadata": self.metadata,
+        }
+
+    def get_metadata(self):
+        metadata = pd.read_csv(
+            self.requires()["metadata"].workdir.joinpath(
+                self.requires()["metadata"].outfile
+            )
+        )
+        return metadata
+
+    @staticmethod
+    def get_stats(audiofiles):
+        """Product the summary of a list of audio directory as dict"""
+        audio_stats = list(map(audio_util.get_audiostats, audiofiles))
+        durations = [stats["duration"] for stats in audio_stats]
+        sample_rates = [stats["sample_rate"] for stats in audio_stats]
+
+        stats = {
+            "total_audiofiles": len(durations),
+            "mean_samplerate": np.mean(sample_rates),
+            "mean_dur(sec)": np.mean(durations),
+            "median_dur)sec)": np.median(durations),
```

Comment:
mean samplerate isn't super meaningful -- I'd way rather have a list of samplerates to see which ones are present in a dataset. For the data before preprocessing having a list of formats would be cool too. There is def a chance that soundfile might not be able to read some of the formats that are present in the dataset though. (looking at you coughvid)

Reply:
Yes. This is a problem. In general, what will be a good way for the `audio_stats` function? Currently renamed to `audio_stats_wav` though. According to #104 . Maybe `ffmpeg` can help, but didnot get the stats directly for a `.webm` file. The following line - https://stackoverflow.com/questions/34118013/how-to-determine-webm-duration-using-ffprobe. The first comment talks about repackaging, please let me know if you already know any alternatives. Thanks.